Sources/SnapshotTesting/Strategy/View.swift

```diff
@@ -0,0 +1,213 @@
+#if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
+import SceneKit
+import SpriteKit
+import WebKit
+import WKSnapshotConfigurationShim
+
+#if os(macOS)
+extension Strategy where A == NSView, B == NSImage {
+  public static var view: Strategy {
+    return .view(precision: 1)
+  }
+
+  public static func view(precision: Float) -> Strategy {
+    return Strategy<NSImage, NSImage>.image(precision: precision).asyncPullback { view in
+      return view.snapshot ?? Async { callback in
+        addImagesForRenderedViews(view).sequence().run { views in
+          let image = NSImage(data: view.dataWithPDF(inside: view.bounds))!
+          let scale = NSScreen.main!.backingScaleFactor
+          image.size = .init(width: image.size.width * 2.0 / scale, height: image.size.height * 2.0 / scale)
+          callback(image)
+          views.forEach { $0.removeFromSuperview() }
+        }
+      }
+    }
+  }
+}
+
+extension Strategy where A == NSView, B == String {
+  public static var recursiveDescription: Strategy<NSView, String> {
+    return SimpleStrategy<String>.lines.pullback { view in
+      return purgePointers(
+        view.perform(Selector(("_subtreeDescription"))).retain().takeUnretainedValue()
+          as! String
+      )
+    }
+  }
+}
+
+extension NSView: DefaultDiffable {
+  public static let defaultStrategy: Strategy<NSView, NSImage> = .view
+}
+#elseif os(iOS) || os(tvOS) || os(watchOS)
+extension Strategy where A == UIView, B == UIImage {
+  public static var view: Strategy {
+    return self.view(precision: 1)
+  }
+
+  public static func view(precision: Float) -> Strategy {
+    return Strategy<CALayer, UIImage>.layer(precision: precision).asyncPullback { view in
+      Async { callback in
+        addImagesForRenderedViews(view).sequence().run { views in
+          callback(view.layer)
+          views.forEach { $0.removeFromSuperview() }
+        }
+      }
+    }
+  }
+}
+
+extension Strategy where A == UIView, B == String {
+  public static var recursiveDescription: Strategy<UIView, String> {
+    return SimpleStrategy<String>.lines.pullback { view in
+      return purgePointers(
+        view.perform(Selector(("recursiveDescription"))).retain().takeUnretainedValue()
+          as! String
+      )
+    }
+  }
+}
+
+extension UIView: DefaultDiffable {
+  public static let defaultStrategy: Strategy<UIView, UIImage> = .view
+}
+#endif
+
+#if os(macOS)
+import Cocoa
+private typealias Image = NSImage
+private typealias ImageView = NSImageView
+private typealias View = NSView
+#elseif os(iOS) || os(tvOS) || os(watchOS)
+import UIKit
+private typealias Image = UIImage
+private typealias ImageView = UIImageView
+private typealias View = UIView
+#endif
+
+private func addImagesForRenderedViews(_ view: View) -> [Async<View>] {
```

Comment:
This traverses a view hierarchy, inserting and collecting image views in an asynchronous fashion (so they can be cleaned up and removed later).